lib/active_merchant/billing/gateways/priority.rb

```diff
@@ -0,0 +1,364 @@
+          card_details['expiryMonth'] = credit_card['expiryMonth']
+          card_details['expiryYear'] = credit_card['expiryYear']
+          card_details['hasContract'] = credit_card['hasContract']
+          card_details['isCorp'] = credit_card['isCorp']
+          card_details['isDebit'] = credit_card['isDebit']
+          card_details['last4'] = credit_card['last4']
+          card_details['token'] = credit_card['token']
+        end
+      end
+
+      def expdate(credit_card)
+        "#{format(credit_card.month, :two_digits)}/#{format(credit_card.year, :two_digits)}"
+      end
+
+      def purchases
+        [{ taxRate: '0.0000', additionalTaxRate: nil, discountRate: nil }]
+      end
+
+      def add_type_merchant_purchase(params, merchant, is_settle_funds, options)
+        params['cardPresent'] = false
+        params['cardPresentType'] = 'CardNotPresent'
+        params['isAuth'] = true
+        params['isSettleFunds'] = is_settle_funds
+        params['isTicket'] = false
+
+        params['merchantId'] = merchant
+        params['mxAdvantageEnabled'] = false
+        params['mxAdvantageFeeLabel'] = ''
+        params['paymentType'] = 'Sale'
+        params['bankAccount'] = nil
+
+        params['purchases'] = purchases
+
+        params['shouldGetCreditCardLevel'] = true
+        params['shouldVaultCard'] = true
+        params['source'] = 'Spreedly'
+        params['sourceZip'] = options[:billing_address][:zip]
+        params['taxExempt'] = false
+        params['tenderType'] = 'Card'
+        params['terminals'] = []
+      end
+
+      def add_type_merchant_refund(params, options)
+        params['cardPresent'] = options['cardPresent']
+        params['clientReference'] = options['clientReference']
+        params['created'] = options['created']
+        params['creatorName'] = options['creatorName']
+        params['currency'] = options['currency']
+        params['customerCode'] = options['customerCode']
+        params['enteredAmount'] = options['amount']
+        params['id'] = 0
+        params['invoice'] = options['invoice']
+        params['isDuplicate'] = false
+        params['merchantId'] = @options[:merchant_id]
+        params['paymentToken'] = options['cardAccount']['token'] if options['cardAccount']
+
+        params['posData'] = options['posData']
+
+        params['purchases'] = options['purchases']
+
+        params['reference'] = options['reference']
+        params['replayId'] = nil
+        params['requireSignature'] = false
+        params['reviewIndicator'] = nil
+
+        params['risk'] = options['risk']
+
+        params['settledAmount'] = options['settledAmount']
+        params['settledCurrency'] = options['settledCurrency']
+        params['settledDate'] = options['created']
+        params['shipToCountry'] = options['shipToCountry']
+        params['shouldGetCreditCardLevel'] = options['shouldGetCreditCardLevel']
+        params['source'] = 'Spreedly'
```

Comment:
Because ActiveMerchant is a collaborative open-source project, this value should come from the `options` hash.